iModelCore/ECDb/ECDb/ECDbImpl.cpp

```diff
@@ -233,7 +238,18 @@ void ECDb::Impl::OnInit() const
     RegisterECSqlPragmas();
     }
 
+//--------------------------------------------------------------------------------------
+// @bsimethod
+//---------------+---------------+---------------+---------------+---------------+------
+DbResult ECDb::Impl::ExecuteDDL(Utf8CP ddl) const {

```

Comment:
Seems a very counter intuitive place to put this logic. I was wondering why so many calls changes from ECDb::ExecuteDdl() to this method. Why not bake this logic directly into the ExecuteDdl() method?

There is no guarantee callers will use this instead of the method on ECDb. Actually, it's more likely people will directly call the other method and not this one. It also seems strange to directly call an Impl method which in turn will call back through the public ECDb interface again...

Reply:
I am curious about this as well @khanaffan.

Reply:
_I thought i answered this few days back but I guess I did not pressed comment._

The objective is to utilize the metadata stored in the ec_* table to update the database schema once a schema changeset has been applied and schema synchronization is active. The solution I've implemented involves disabling DDL tracking during the database schema synchronization post-application of a schema changeset.

As a principle, change tracking should never be disabled unless applying an incoming changeset. This is precisely why the current change tracker in BeSQlite::Db is not accessible via a function. It can only be enabled or disabled within the TxnManager, which acts as a change tracker itself. Therefore, we should not introduce a method in Db to disable it, preventing any assumptions by programmers that it can be disabled at will.

My implementation aims to restrict access to ECDb for a very specific case, and even within ECDb, tracking cannot be disabled unless the `DisableDdlTracking` scope is used on the stack. Therefore, my main goal is to narrow the circumstances under which DDL tracking can be disabled, ensuring that any attempt to do so will be evident during a peer review.